adapters/grid/grid.go

```diff
@@ -69,16 +74,41 @@ func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
 	return imp
 }
 
+func setImpExtKeywords(imp openrtb2.Imp, request *openrtb2.BidRequest) error {
+	var ext adapters.ExtImpBidder
+	var gridExt openrtb_ext.ExtImpGrid
+	var reqExt ReqExt
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &gridExt); err != nil {
+		return err
+	}
+	if gridExt.Keywords != nil {
+		if request.Ext != nil {
+			if err := json.Unmarshal(request.Ext, &reqExt); err != nil {
+				return err
+			}
+		}
+
+		reqExt.Keywords = gridExt.Keywords
```

Comment:
I realize the `root.ext` json usually doesn't come with a `keywords` field but, should we check whether or not it is empty before rewriting? Is there any possibility that someone might slip some keywords in there that we might be interested in?

Reply:
Hello! We have fixed this case + added keywords format validation